buildpacks/nodejs-function-invoker/src/main.rs

```diff
@@ -60,27 +65,44 @@ impl Buildpack for NodeJsInvokerBuildpack {
     fn build(&self, context: BuildContext<Self>) -> libcnb::Result<BuildResult, Self::Error> {
         log_header("Heroku Node.js Function Invoker Buildpack");
 
+        let app_dir = &context.app_dir;
+        let metadata_runtime = &context.buildpack_descriptor.metadata.runtime;
+        let package_name = &metadata_runtime.package_name;
+        let package_version = &metadata_runtime.package_version;
+
         log_info("Checking for function file");
-        get_main(&context.app_dir).map_err(NodeJsInvokerBuildpackError::MainFunctionError)?;
-
-        context.handle_layer(
-            layer_name!("runtime"),
-            RuntimeLayer {
-                package: context
-                    .buildpack_descriptor
-                    .metadata
-                    .runtime
-                    .package
-                    .clone(),
-            },
-        )?;
+        get_main(app_dir).map_err(NodeJsInvokerBuildpackError::MainFunctionError)?;
+
+        let declared_runtime_package = get_declared_runtime_package(app_dir, package_name)
+            .map_err(NodeJsInvokerBuildpackError::ExplicitRuntimeDependencyFunctionError)?;
+
+        if let Some((package_name, package_version)) = declared_runtime_package {
+            log_info(format!(
+                "Runtime declared in package.json: {0}@{1}",
+                package_name.clone(),
+                package_version
+            ));
+        } else {
+            log_warning(
+                "Deprecation",
+                format!("Future versions of the Functions Runtime for Node.js ({0}) will not be auto-detected \
+                and must be added as a dependency in package.json.", package_name)
+            );
+            context.handle_layer(
+                layer_name!("implicit_runtime"),
+                RuntimeLayer {
+                    package: format!("{0}@{1}", package_name, package_version),
+                },
+            )?;
+        }
 
         BuildResultBuilder::new()
             .launch(
                 LaunchBuilder::new()
                     .process(
-                        ProcessBuilder::new(process_type!("web"), "sf-fx-runtime-nodejs")
+                        ProcessBuilder::new(process_type!("web"), "npx")
                             .args(vec![
+                                "sf-fx-runtime-nodejs",
```

Comment:
Should this be dynamic instead? It could either be just the package name as read from `buildpack.toml`, or it could be explicitly the package name and version we found in their `package.json`?

Reply:
I'm actually surprised this works at all. Don't we need run `npx @heroku/sf-fx-runtime-nodejs` rather than just `npx sf-fx-runtime-nodejs`?

Reply:
i'd have to split up the package name from `@heroku/sf-fx-runtime-nodejs` into org/name parts which is easy enough to do.  

it's technically not the name of the package required here though, rather the [`bin` entry](https://github.com/forcedotcom/sf-fx-runtime-nodejs/blob/main/package.json#L69) we specify in `@heroku/sf-fx-runtime-nodejs`.  does that change your opinion here at all?

Reply:
nvm, did some more reading about how `npx` determines the executable and using the full package name is preferred vs. relying on the heuristics used [when no `--package` options are provided](https://docs.npmjs.com/cli/v8/commands/npx#description).  will fix this.